src/runtimes/docker/clients/DockerClientBase/DockerClientBase.ts

```diff
@@ -352,7 +353,7 @@ export abstract class DockerClientBase extends ConfigurableClient implements ICo
                     : options.disableContentTrust),
             withDockerLabelsArg(options.labels),
             withNamedArg('--iidfile', options.imageIdFile),
-            withNamedArg('--platform', options.platform),
+            withNamedArg('--platform', getPlatformString(options?.platform)),

```

Comment:
Consider creating a new custom arg method `withDockerPlatformArg(options.platform)` (no need for ? on options, since it isn't valid for options to be null or undefined). along the lines of withDockerBuildArg.